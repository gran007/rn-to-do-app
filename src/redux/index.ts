import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';

import { reducer as commonReducer, actions as commonActions } from './reducers/Common';
import { reducer as authReducer, actions as authActions } from './reducers/Auth';
const actions = {
  auth: authActions,
  common: commonActions,
}
export { actions };


export const store = configureStore({
  reducer: combineReducers({
    common: commonReducer,
    auth: authReducer,
  })
});

export type RootState = ReturnType<typeof store.getState>;
