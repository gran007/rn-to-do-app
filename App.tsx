import React from 'react';
import { Provider } from 'react-redux';
import { store } from '@todo/redux'
import Navigator from '@todo/app-navigator'


export default function App() {
  return (
    <Provider store={store}>      
      <Navigator />
    </Provider>
  )
};
