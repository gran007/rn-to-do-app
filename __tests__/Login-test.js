/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import Login from '@todo/pages/01.login';
import { render, fireEvent } from "@testing-library/react-native";
import Strings from '@todo/strings'
import { store } from '@todo/redux'
import { Provider } from 'react-redux';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const rendered = renderer.create(<App />);
  expect(rendered).toBeTruthy();
});

it('App Snapshot', () => {
  const snap = renderer.create(<App />).toJSON();
  expect(snap).toMatchSnapshot();
});

const getElement = () => {
  const { getByTestId } = render(
    <App/>
  );
  const emailInput = getByTestId('emailInput');
  const errorText = getByTestId('errorText');
  const passwordInput = getByTestId('passwordInput');
  const loginButton = getByTestId('loginButton');
  return { emailInput, errorText, passwordInput, loginButton };
}

describe("로그인 테스트", () => {
  
  it('이메일 유효성 실패 검증', () => {
    const { emailInput, errorText } = getElement();
    fireEvent(emailInput, 'onChangeText', 'test@test');
    expect(errorText.props.children).toBe(Strings.EMAIL_ERROR)
  })

  it('이메일 유효성 검증', () => {
    const { emailInput, errorText } = getElement();
    fireEvent(emailInput, 'onChangeText', 'test@test.com');
    expect(errorText.props.children).toBe('')
  })

  it('로그인 실패', () => {
    const { emailInput, errorText, passwordInput, loginButton } = getElement();
    fireEvent(emailInput, 'onChangeText', 'test@test.com');
    fireEvent(passwordInput, 'onChangeText', '123');
    fireEvent(loginButton, 'onPress');
    expect(errorText.props.children).toBe(Strings.LOGIN_ERROR)
  })

  it('로그인 성공', () => {
    const { emailInput, errorText, passwordInput, loginButton } = getElement();
    fireEvent(emailInput, 'onChangeText', 'test@test.com');
    fireEvent(passwordInput, 'onChangeText', '1234');
    fireEvent(loginButton, 'onPress');
    // expect(errorText.props.children).toBe(Strings.LOGIN_ERROR)
  })

});

