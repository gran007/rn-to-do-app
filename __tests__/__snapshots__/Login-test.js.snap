// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      keyboardShouldPersistTaps="handled"
      showsVerticalScrollIndicator={false}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "marginBottom": 66.7,
                "marginTop": 100.05000000000001,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "NotoSansKR-Bold",
                  "fontSize": 32,
                  "marginBottom": 33.35,
                }
              }
            >
              Simple Todo
            </Text>
            <Image
              source={
                Object {
                  "testUri": "../../../src/images/logo.png",
                }
              }
              style={
                Object {
                  "height": 154.16666666666669,
                  "width": 154.16666666666669,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <TextInput
              keyboardType="email-address"
              onChangeText={[Function]}
              onFocus={[Function]}
              style={
                Object {
                  "borderColor": "#000000",
                  "borderRadius": 16.666666666666668,
                  "borderWidth": 1,
                  "color": "#000",
                  "fontFamily": undefined,
                  "fontSize": 18,
                  "height": 108.33333333333334,
                  "marginBottom": 50.025000000000006,
                  "width": 600,
                }
              }
              testID="emailInput"
              value=""
            />
            <View
              style={
                Object {
                  "left": 16.666666666666668,
                  "position": "absolute",
                  "top": 108.3875,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#ff3b30",
                    "fontFamily": undefined,
                    "fontSize": 14,
                  }
                }
                testID="errorText"
              >
                
              </Text>
            </View>
          </View>
          <TextInput
            onChangeText={[Function]}
            onFocus={[Function]}
            secureTextEntry={true}
            style={
              Object {
                "borderColor": "#000000",
                "borderRadius": 16.666666666666668,
                "borderWidth": 1,
                "color": "#000",
                "fontFamily": undefined,
                "fontSize": 18,
                "height": 108.33333333333334,
                "marginBottom": 50.025000000000006,
                "width": 600,
              }
            }
            testID="passwordInput"
            value=""
          />
          <View
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#c8c7cc",
                "borderRadius": 16.666666666666668,
                "height": 108.33333333333334,
                "justifyContent": "center",
                "marginBottom": 66.7,
                "marginTop": 33.35,
                "opacity": 1,
                "width": 600,
              }
            }
            testID="loginButton"
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": undefined,
                  "fontSize": 18,
                }
              }
            >
              로그인
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>,
  <View />,
]
`;
